### PR TITLE
support absolute path in .bowerrc directory option

### PR DESCRIPTION
**Pull request by gronke**
As discussed in #1914 we want to support absolute paths in `.bowerrc` directory option. A simple check if the directory begin with a `/` decides whether the path is relative and prefixed with the current work directory or taken as absolute path.

### CODE REVIEW

**Comment by sheerun** [comment]
Hey. `Promise` is not available in node 0.10. We use Q library for promises.

Could you fix it and squash the changes?

**Comment by gronke** [comment]
Yep, saw that and will switch to Q.

File: lib/core/Manager.js [hunk 1]
```diff
@@ -124,7 +125,7 @@ Manager.prototype.resolve = function () {
 
 Manager.prototype.preinstall = function (json) {
     var that = this;
-    var componentsDir = path.join(this._config.cwd, this._config.directory);
```
Comment (sheerun) [inline lib/core/Manager.js]: Maybe we can use `path.resolve` instead? to support relative config.directory

**Comment by sheerun** [comment]
I think there are some tests missing, because you did not change:

- https://github.com/gronke/bower/blob/enhancement/absolute-paths-1914/lib/core/Manager.js#L638
- https://github.com/bower/bower/blob/master/lib/commands/link.js#L52

File: lib/core/Manager.js (hunk 1 above)
Comment (gronke) [inline lib/core/Manager.js]: When the config.directory begins with an `/` it's expected to be absolute. In any other case the path is expected to be relative and joined with `config.cwd`. Not sure if I understood your comment correctly. How can `path.resolve` help here?

Comment (sheerun) [inline lib/core/Manager.js]: resolve "removes" `..` from joined path.

path.join('/foo/bar', '../baz') == '/foo/bar/../baz'
path.resolve('/foo/bar', '../baz') == '/foo'

Comment (gronke) [inline lib/core/Manager.js]: Done

**Comment by sheerun** [comment]
I think it's good now. Thank you for being patient

**Pull request merged**